shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -320,27 +320,37 @@ - (void)dispatchPointerDataPacket:(std::unique_ptr<flutter::PointerDataPacket>)p
 }
 
 - (fml::WeakPtr<flutter::PlatformView>)platformView {
-  FML_DCHECK(_shell);
+  if (!_shell) {
+    return {};
+  }
   return _shell->GetPlatformView();
 }
 
 - (flutter::PlatformViewIOS*)iosPlatformView {
-  FML_DCHECK(_shell);
+  if (!_shell) {
+    return nullptr;
```

Comment:
curious why we use WeakPtr in previous line, but raw pointer here?

Reply:
and `RefPtr` below? (Should they be `WeakPtr` as well?)

Reply:
use cases: 
* `WeakPtr`: A raw pointer wrapper that, in unopt builds, asserts that all dereferences occur on the thread on which it was allocated. We use this to ensure that all derefs of `shell` occur on the platform thread. Does not do refcounting. We use this where we want to enforce threading invariants but object ownership/lifetime is managed by someone else.
* `RefPtr`: A refcounted pointer. We use this in the engine where an object ownership is shared between a pool of owners.
* raw C/C++ pointers: In cases where we want a non-owning pointer to an object whose lifetime is managed by someone else and where we don't care about threading invariants, we often use raw C/C++ pointers.

When I wrote this patch, I looked and these getters are only really used by FlutterEngine itself, FlutterViewController, and tests. There's no reason we couldn't be consistent and return at raw pointer instead of the WeakPtr in the `platformView` getter, but that's orthogonal to this fix. 

My thinking is that rather than do that though, I'd rather send a followup that eliminates these getters altogether and provides better API instead. For example, `engine.iosPlatformView` is used by accessibility code in `FlutterViewController`; the relevant non-view-specific code should simply be refactored into FlutterEngine instead.